src/lib/radio/radio.ts

```diff
@@ -400,13 +400,21 @@ export class MatRadioButton extends _MatRadioButtonMixinBase
   set value(value: any) {
     if (this._value != value) {
       this._value = value;
+      this._markForCheck();
       if (this.radioGroup != null) {
         if (!this.checked) {
           // Update checked when the value changed to match the radio group's value
           this.checked = this.radioGroup.value == value;
         }
         if (this.checked) {
+          const groupValueChanged = this.radioGroup.value !== this.value;
           this.radioGroup.selected = this;
+          if (groupValueChanged) {
+            setTimeout(() => {
+              this.radioGroup._controlValueAccessorChangeFn(this.value);
+              this.radioGroup._emitChangeEvent();
```

Comment:
Should we try to move this kind of logic into a function? I feel like it can be simplified and kept more clean..

https://github.com/tinayuangao/material2/blob/6e0293f8210584296077d16d36cff3e9bcb175a8/src/lib/radio/radio.ts#L589-L599

Reply:
Thanks for review. 

Updated.